src/service/atm_service.rb

```diff
@@ -0,0 +1,134 @@
+require_relative('../model/atm_unit.rb')
+require_relative('./transaction_service.rb')
+
+require 'date'
+
+module AtmService
+
+    def AtmService.deposit_money(amount, bank_account, atm_unit)
+        bank_account.amount += amount
+        atm_unit.total_amount += amount
+    end
+
+    def AtmService.withdraw_money(amount, bank_account, atm_unit)
+        # sort transactions by date
+        TransactionService.sort_by_date(bank_account.transactions) 
+
+        # check if there are any withdraw transactions in the last 24 hours since the current transaction
+        total_withdraw = 0
+        current_datetime = Time.now
+        bank_account.transactions.each do |transaction|
+            if ((current_datetime - transaction.datetime) / 3600).round <= 24 && transaction.amount < 0
+                total_withdraw += transaction.amount
+            end
+        end
+
+        if total_withdraw.abs > 5000
```

Comment:
I think a constant for this value would help. If we change the limit from 5000 to 10000, you need to change a lot of code :D